### ISSUE TITLE
Document what a set of password requirements not worthy of incorporation into the project is

### ISSUE THREAD

**Issue by rmondello** [apple/password-manager-resources#134]
In https://github.com/apple/password-manager-resources/pull/86, David Jennes writes:

> This is maybe a more general discussion for this project: do we want to document every password quirk for each website? Or can we agree to a set of rules that we assume (or require?) that each password manager will at least adhere to? I'm pretty sure that each password manager will generate something longer than 8 characters. Not sure if the same can be said for 16 chars (or longer), that's something that may be dependent on the user's settings.

This seems worth doing, to me.

**Comment by geoffcline** [comment]
I think microsoft's documentation for windows 10 is a relevant example of "default password requirements"

https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/password-must-meet-complexity-requirements#reference

In relevant part:

> The password contains characters from three of the following categories:
> - Uppercase letters of European languages (A through Z, with diacritic marks, Greek and Cyrillic characters)
> - Lowercase letters of European languages (a through z, sharp-s, with diacritic marks, Greek and Cyrillic characters)
> - Base 10 digits (0 through 9)
> - Non-alphanumeric characters (special characters): (~!@#$%^&*_-+=`|\(){}[]:;"'<>,.?/) Currency symbols such as the Euro or British Pound are not counted as special characters for this policy setting.
> - Any Unicode character that is categorized as an alphabetic character but is not uppercase or lowercase. This includes Unicode characters from Asian languages.

The Azure AD (msft's cloud directory product) documentation isn't very helpful. It only talks about min length and a generic "is complex enough" standard.

https://docs.microsoft.com/en-us/azure/active-directory-domain-services/password-policy

GSuite is very similar

- https://imgur.com/a/ySx60OA 

- google's docs on "secure passwords" https://support.google.com/accounts/answer/9094506?hl=en

Okta mentions a default password policy -- but i couldn't find it online (https://help.okta.com/en/prod/Content/Topics/Security/Security_Policies.htm). 

I think microsoft's policy for windows 10 could be a good starting point.

**Comment by BMichael00-alt** [comment]
Yeah if I could get it back then I'd be able to see the results of my life work on the next three weeks or so to get the best quality of it in my own way and my work is not just about the sexual aspect but the fact that I can do that with the other girls is just as important as a member of the time to get it done with a little bit more than I can do with my own way of doing this for me and u know what I'm trying to say that i'm a good team in my room with a lot on the road to get it done with a smile

**Comment by jpgoldberg** [comment]
As elsewhere, we need to recognize the distinction between policy meant for humans and policy meant for password generators. 

Given what our generator, any site that allows uppers, lowers, digits and any of the symbols `-!@._*` will be happy with what we create:

So our defaults are happy with any subset (including a proper subset) of the set of passwords matching

```
allow: upper;
allow: lower;
allow: digit;
allow: [-!@._*];
```
with maxsize 20 or greater.

Our default will get along fine with any site that switches any (or all) of those `allowed` to `required`.

Note that we seem to switch back and forth about putting the those symbols in by default.  So we wouldn't mind terribly if what is done here just says anything that is a subset of the set of passwords conforming to

```
allow: upper;
allow: lower;
allow: digit;
```
 
Again with maxsize >= 20.

**Comment by rmondello** [comment]
Thanks @jpgoldberg! Can you speak to how the set of characters for `allow: [-!@._*];` was derived? Did you find something special about those characters?

iCloud Keychain’s defaults are happy with any subset of passwords matching:
```
allow: upper;
allow: lower;
allow: digit;
allow: [-];
```
allowing a length of 20.

**Comment by jpgoldberg** [comment]
> Can you speak to how the set of characters for allow: [-!@._*]; was derived?

Yes you may. Discussion started in

   https://github.com/1Password/spg/issues/14

and continued in 

   https://github.com/1Password/spg/pull/22

**Comment by jpgoldberg** [comment]
Following up on a private conversation, I think that trying to select which rules to include in this data set counts as the evil sort of premature optimization. 

Instead we should anticipate that all of our defaults are things that we will be tinkering with as time goes on, and so if there is no real barrier or prohibitive cost to just including every rule set we know, we should probably just collect them all with no pre-filtering on relevance.
